wurst/lib/Lodash.wurst

```diff
@@ -305,6 +316,341 @@ public function each<T>(VoidFunction<T> func, LinkedList<T> list)
+        let lst = new LinkedList<int>()
+        for x from this
+            lst.add(x)
+        close()
+        return lst
+
+public function range(int min, int max, int incr) returns Range
+    return new Range(min, max, incr)
+
+public function range(int min, int max) returns Range
+    return new Range(min, max, 1)
+
+public function range(int max) returns Range
+    return new Range(0, max, 1)
+
+public function range() returns Range
+    return new Range()
+
+public function rangeStep(int max, int incr) returns Range
+    return new Range(0, max, incr)
+
+public function uniq<T>(LinkedList<T> lst) returns LinkedList<T>
+    let stored = new HashSet<T>()
+    let uniqLst = new LinkedList<T>()
+    let iter = lst.iterator()
+    for elem from iter
+        if not stored.has(elem)
+            uniqLst.push(elem)
+            stored.add(elem)
+    iter.close()
+    lst.maybeFree()
+    destroy stored
+    return uniqLst
+
+public function LinkedList<T>.uniq<T>() returns LinkedList<T>
+    return uniq(this)
+
+public function uniqBy<T, R>(LinkedList<T> lst, Function<T, R> func) returns LinkedList<T>
+    let stored = new HashSet<R>()
+    let uniqLst = new LinkedList<T>()
+    let iter = lst.iterator()
+    for elem from iter
+        let res = func.call(elem)
+        if not stored.has(res)
+            stored.add(res)
+            uniqLst.push(elem)
+    iter.close()
+    lst.maybeFree()
+    func.maybeFree()
+    destroy stored
+    return uniqLst
+
+public function LinkedList<T>.uniqBy<T, R>(Function<T, R> func) returns LinkedList<T>
+    return uniqBy(this, func)
+
+public function union<T>(LinkedList<T> a, LinkedList<T> b) returns LinkedList<T>
+    let stored = new HashSet<T>()
+    let lst = new LinkedList<T>()
+    let iterA = a.iterator()
+    let iterB = b.iterator()
+    for elem from iterA
+        if not stored.has(elem)
+            stored.add(elem)
+            lst.push(elem)
+    for elem from iterB
+        if not stored.has(elem)
+            stored.add(elem)
+            lst.push(elem)
+    iterA.close()
+    iterB.close()
+    a.maybeFree()
+    b.maybeFree()
+    destroy stored
+    return lst
+
+public function LinkedList<T>.union<T>(LinkedList<T> lst) returns LinkedList<T>
+    return union(this, lst)
+
+public function intersection<T>(LinkedList<T> a, LinkedList<T> b) returns LinkedList<T>
+    let stored = new HashSet<T>()
+    let lst = new LinkedList<T>()
+    let iterA = a.iterator()
+    let iterB = b.iterator()
+    for elem from iterA
+        stored.add(elem)
+    for elem from iterB
+        if stored.has(elem)
+            lst.push(elem)
+    iterA.close()
+    iterB.close()
+    a.maybeFree()
+    b.maybeFree()
+    destroy stored
+    return lst
+
+public function LinkedList<T>.intersection<T>(LinkedList<T> lst) returns LinkedList<T>
+    return intersection(this, lst)
+
+public function difference<T>(LinkedList<T> a, LinkedList<T> b) returns LinkedList<T>
+    T array occurences
```

Comment:
Why do you use an array here instead of a HashSet?

Reply:
@jonathanmclaus good point, I've switched it to a HashSet. Array is more performant though so I might switch it back for most types and use the `HashSet` for primitive types (other than int and maybe bool) and handles.

Reply:
Are you going to do that now or should I accept? Not really clear on the difference between the built-in `array` and `initHashtable`.

Reply:
Should be updated

Reply:
Ah I meant I'll leave it for a later PR